addons/website_profile/controllers/main.py

```diff
@@ -259,16 +257,38 @@ def view_all_users_page(self, page=1, **kwargs):
                 user['karma_gain'] = user_data.get('karma_gain_total', 0)
             user_values.sort(key=itemgetter('position'))
 
+            if (uid not in users.ids):
+                # Need to keep the dom to search only for users that appear in the ranking page
+                current_user = User.sudo().search(expression.AND([[('id', '=', uid)], dom]))
```

Comment:
you most than probably get user in cache
-> if user.karma and user.website_published

1 extra sql query won ;)

Reply:
how do you do that ? I thought the search would do that, put the data in the cache.
I guess you want me to do that only when user.karma and user.website_published = true.

Reply:
```suggestion
                current_user = User.sudo().search(expression.AND([[('id', '=', uid)], dom]))
```
if uid not in users.ids and request.env.user.karma and request.env.user.website_published